solr/core/src/java/org/apache/solr/rest/RestManager.java

```diff
@@ -315,6 +315,16 @@ public void doInit(SolrQueryRequest solrRequest, SolrQueryResponse solrResponse)
       if (managedResource == null) {
         final String method = getSolrRequest().getHttpMethod();
         if ("PUT".equals(method) || "POST".equals(method)) {
+          // check for typos in the feature-store or model-store endpoints (like features-store or
+          // models-store)
+          if (restManager.managed.containsKey("/schema/feature-store")
+              || restManager.managed.containsKey("/schema/model-store")) {
+            if (!restManager.managed.containsKey(resourceId)) {
```

Comment:
Maybe I would invert the if statement... you want to check for "feature-store"/"model-store" only if you do not have the resourceId, right?

Reply:
yes, ok